faker.go

```diff
@@ -325,6 +331,11 @@ func SetRandomNumberBoundaries(start, end int) error {
 // FakeData is the main function. Will generate a fake data based on your struct.  You can use this for automation testing, or anything that need automated data.
 // You don't need to Create your own data for your testing.
 func FakeData(a interface{}) error {
+	// if parameter != pre-parameter reset stateful data
+	if a != currentFaker {
+		currentFaker = a
+		boundaryCurrentNumber = 0
+	}
```

Comment:
Isn't this going to be not safe for multi-thread processing?

Reply:
fixed

Reply:
Incr/Decr does not support multi-thread and multi-struct thread-safe operation, is a stateful processing.
Example struct A and B
fakeData(A) -> 1
fakeData(A) -> 2
fakeData(B) -> 1
fakeData(A) -> 1, excepted 3
fakeData(B) -> 1, excepted 2
I need to record all procedure data or lock n-m fakeData processing, i think that would be too complicated and ineffective.

Single struct and multi-thread operation locked by func stepIntegerWithBoundary (in facker.go line 1150-1152)
func stepIntegerWithBoundary(boundary numberBoundary) (int, error) {
	mu.Lock()
	defer mu.Unlock()